File: MatriX/samples/csharp/MiniClient/FrmVCard.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Matrix;
using Matrix.Xmpp.Client;
using Matrix.Xmpp.Vcard;

namespace MiniClient
{
    public partial class FrmVCard : Form
    {
        private readonly XmppClient xmppClient;
      

        public FrmVCard(XmppClient xc, Jid jid, bool own)
        {
            InitializeComponent();

            xmppClient = xc;

            if (own)
            {
                Text = "My VCard";
                cmdPublish.Enabled = true;
                GetMyVcard();
            }
            else
            {
                Text = "VCard: " + jid;
                GetVcard(jid);
            }
                
        }

        private void GetVcard(Jid jid)
        {
            var viq = new VcardIq {To = jid, Type = Matrix.Xmpp.IqType.get};
            xmppClient.IqFilter.SendIq(viq, VcardResponse);
        }

        private void GetMyVcard()
        {
            var viq = new VcardIq {Type = Matrix.Xmpp.IqType.get };
            xmppClient.IqFilter.SendIq(viq, VcardResponse);
        }

        private void VcardResponse(object sender, IqEventArgs e)
        {
            if (e.Iq.Type == Matrix.Xmpp.IqType.result)
            {
                var vc = e.Iq.Query as Vcard;
                txtName.Text = vc.Fullname;
                txtNickname.Text = vc.Nickname;

                var  email = vc.GetEmails().FirstOrDefault(m => m.IsInternet);
                if (email != null)
                    txtEmail.Text = email.Address;
            }
        }


        private void SetMyVcard()
        {
            var viq = new VcardIq { Type = Matrix.Xmpp.IqType.set };
            
            if (!string.IsNullOrEmpty(txtName.Text))
                viq.Vcard.Fullname = txtName.Text;

            if (!string.IsNullOrEmpty(txtNickname.Text))
            viq.Vcard.Nickname = txtNickname.Text;

            if (!string.IsNullOrEmpty(txtEmail.Text))
                viq.Vcard.AddEmail(new Email {IsInternet = true, Address = txtEmail.Text});
            
            xmppClient.IqFilter.SendIq(viq, VcardUpdateResponse);
        }

        private void VcardUpdateResponse(object sender, IqEventArgs e)
        {
            if (e.Iq.Type == Matrix.Xmpp.IqType.result)
            {
                MessageBox.Show("The VCard was updated successful!!");
            }
        }

        private void cmdPublish_Click(object sender, System.EventArgs e)
        {
            SetMyVcard();
        }
    }
}

File: MatriX/samples/vb/MiniClient/FrmVCard.vb
Imports System.Linq
Imports System.Windows.Forms
Imports Matrix
Imports Matrix.Xmpp.Client
Imports Matrix.Xmpp.Vcard

Public Partial Class FrmVCard
    Inherits Form
    Private ReadOnly xmppClient As XmppClient


    Public Sub New(xc As XmppClient, jid As Jid, own As Boolean)
        InitializeComponent()

        xmppClient = xc

        If own Then
            Text = "My VCard"
            cmdPublish.Enabled = True
            GetMyVcard()
        Else
            Text = "VCard: " + jid.ToString()
            GetVcard(jid)

        End If
    End Sub

    Private Sub GetVcard(jid As Jid)
        Dim viq = New VcardIq() With { _
                .[To] = jid, _
                .Type = Xmpp.IqType.[get] _
                }
        xmppClient.IqFilter.SendIq(viq, AddressOf VcardResponse)
    End Sub

    Private Sub GetMyVcard()
        Dim viq = New VcardIq() With { _
                .Type = Xmpp.IqType.[get] _
                }
        xmppClient.IqFilter.SendIq(viq, AddressOf VcardResponse)
    End Sub

    Private Sub VcardResponse(sender As Object, e As IqEventArgs)
        If e.Iq.Type = Xmpp.IqType.result Then
            Dim vc = TryCast(e.Iq.Query, Vcard)
            txtName.Text = vc.Fullname
            txtNickname.Text = vc.Nickname

            Dim email = vc.GetEmails().FirstOrDefault(Function(m) m.IsInternet)
            If email IsNot Nothing Then
                txtEmail.Text = email.Address
            End If
        End If
    End Sub


    Private Sub SetMyVcard()
        Dim viq = New VcardIq() With { _
                .Type = Xmpp.IqType.[set] _
                }

        If Not String.IsNullOrEmpty(txtName.Text) Then
            viq.Vcard.Fullname = txtName.Text
        End If

        If Not String.IsNullOrEmpty(txtNickname.Text) Then
            viq.Vcard.Nickname = txtNickname.Text
        End If

        If Not String.IsNullOrEmpty(txtEmail.Text) Then
            viq.Vcard.AddEmail(New Email() With { _
                                  .IsInternet = True, _
                                  .Address = txtEmail.Text _
                                  })
        End If

        xmppClient.IqFilter.SendIq(viq, AddressOf VcardUpdateResponse)
    End Sub

    Private Sub VcardUpdateResponse(sender As Object, e As IqEventArgs)
        If e.Iq.Type = Xmpp.IqType.result Then
            MessageBox.Show("The VCard was updated successful!!")
        End If
    End Sub

    Private Sub cmdPublish_Click(sender As Object, e As System.EventArgs) Handles cmdPublish.Click
        SetMyVcard()
    End Sub
End Class